pkg/server/wire.go

```diff
@@ -79,6 +80,7 @@ var wireBasicSet = wire.NewSet(
 	routing.ProvideRegister,
 	wire.Bind(new(routing.RouteRegister), new(*routing.RouteRegisterImpl)),
 	hooks.ProvideService,
+	kvstore.ProvideService,
```

Comment:
Nit. ProvideService currently returns an interface. For every other "provider"/dependency we let the provider return the struct pointer and then use `wire.Bind` to bind to an interface instead.  I think we've sort of come to the conclusion that this is how we would like to manage things moving forward, basically _accept interfaces, return structs_.

Reply:
The intent in returning an interface is that ProvideService may in the future return an alternate implementation based on config, eg if we wanted to allow using a different storage backend.  I wanted to make it clear that the caller should not rely on that specific implementation but rather accept anything that implements the interface.  Maybe I'm getting ahead of myself there, what do you think?

Reply:
> I wanted to make it clear that the caller should not rely on that specific implementation

Agree with that. The caller will accept an interface and use that to interact with the store, but deciding what implementation to use will be handled by wire. At least with Enterprise this way allow Enterprise to depend on an OSS service (struct pointer), extending/overriding it with a different implementation. 

But yeah this not a big issue currently since kvstore it not used yet.

Reply:
Ok, I'll merge it as-is and we can always tweak things later 👍